### PR TITLE
Update leader election in prep for changing to multi-read, single-write

### PR DESCRIPTION
**Pull request by youngnick**
Updates #1385 

- Leader election on by default.
- Leader election configmap now `leader-elect` not `contour` (will be created if it doesn't exist)
- Added RBAC for HttpLoadBalancer CRDs.
- Updated `contour` deployment.

Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

File: cmd/contour/serve.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ func registerServe(app *kingpin.Application) (*kingpin.CmdClause, *serveContext)
 	serve.Flag("envoy-service-https-port", "Kubernetes Service port for HTTPS requests").IntVar(&ctx.httpsPort)
 	serve.Flag("use-proxy-protocol", "Use PROXY protocol for all listeners").BoolVar(&ctx.useProxyProto)
 
-	serve.Flag("enable-leader-election", "Enable leader election mechanism").BoolVar(&ctx.EnableLeaderElection)
+	serve.Flag("disable-leader-election", "Enable leader election mechanism").BoolVar(&ctx.DisableLeaderElection)
```
Comment (davecheney) [inline cmd/contour/serve.go]: please fix the description to match the flag

File: cmd/contour/serve.go [hunk 2]
```diff
@@ -300,10 +300,9 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	// managing the close process when the other goroutines are closed.
 	g.AddContext(func(electionCtx context.Context) {
 		log := log.WithField("context", "leaderelection")
-		if !ctx.EnableLeaderElection {
+		if ctx.DisableLeaderElection {
```
Comment (davecheney) [inline cmd/contour/serve.go]: Please add a TODO that we can avoid setting up leadership election if its disable; just close leader OK and don't call AddContext.

Comment (davecheney) [reply cmd/contour/serve.go]: Same as line 312 which I can't add a comment on because github.

File: cmd/contour/serve.go [hunk 3]
```diff
@@ -353,33 +352,26 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
...
-			log.Info("started")
-			defer log.Info("stopped")
-
-			go func() {
-				<-stop
-				s.Stop()
-			}()
-
-			return s.Serve(l)
+		addr := net.JoinHostPort(ctx.xdsAddr, strconv.Itoa(ctx.xdsPort))
```
Comment (davecheney) [inline cmd/contour/serve.go]: please update the comment on line 343

File: cmd/contour/servecontext.go [hunk 4]
```diff
@@ -80,8 +80,8 @@ type serveContext struct {
 	// permitInsecure field in IngressRoute.
 	DisablePermitInsecure bool `yaml:"disablePermitInsecure,omitempty"`
 
-	// EnableLeaderElection should only be set by command line flag.
-	EnableLeaderElection bool `yaml:"-"`
+	// DisableLeaderElection should only be set by command line flag.
```
Comment (davecheney) [inline cmd/contour/servecontext.go]: s/should/can/

File: examples/common/rbac.yaml [hunk 5]
```diff
@@ -58,6 +58,15 @@ rules:
   - put
   - post
   - patch
+- apiGroups: ["projectcontour.io"]
```
Comment (davecheney) [inline examples/common/rbac.yaml]: looks unrelated

File: cmd/contour/serve.go (hunk 1 above)
Comment (youngnick) [reply cmd/contour/serve.go]: Oops.
